honeybee_energy/material/opaque.py

```diff
@@ -638,3 +646,480 @@ def __copy__(self):
+        density: Number for the density of the dry soil [kg/m3].
+        specific_heat: Number for the specific heat of the material [J/kg-K].
+        roughness: Text describing the relative roughness of the material.
+            Must be one of the following: 'VeryRough', 'Rough', 'MediumRough',
+            'MediumSmooth', 'Smooth', 'VerySmooth'. Default is 'MediumRough'.
+        thermal_absorptance: A number between 0 and 1 for the fraction of
+            incident long wavelength radiation that is absorbed by the material.
+            Default value is 0.9.
+        solar_absorptance: A number between 0 and 1 for the fraction of incident
+            solar radiation absorbed by the material. Default value is 0.7.
+        visible_absorptance: A number between 0 and 1 for the fraction of incident
+            visible wavelength radiation absorbed by the material.
+            Default is None, which will yield the same value as solar_absorptance.
+        plant_height: This field defines the height of plants in units of meters. 
+            This field is limited to values in the range 0.005 < Height < 1.00 m. 
+            Default is .2 m.
+        leaf_area_ind: This is the projected leaf area per unit area of soil surface.
+            This field is dimensionless and is limited to values in the range of
+            0.001 < LAI < 5.0. Default is 1.0. At the present time the fraction 
+            vegetation cover is calculated directly from LAI (Leaf Area Index) 
+            using an empirical relation.
+        leaf_reflectivity: This field represents the fraction of incident solar 
+            radiation that is reflected by the individual leaf surfaces (albedo). 
+            Solar radiation includes the visible spectrum as well as infrared 
+            and ultraviolet wavelengths. Values for this field must be between 
+            0.05 and 0.5. Default is .22. Typical values are .18 to .25.
+        leaf_emissivity: This field is the ratio of thermal radiation emitted from 
+            leaf surfaces to that emitted by an ideal black body at the same temperature. 
+            This parameter is used when calculating the long wavelength radiant exchange 
+            at the leaf surfaces. Values for this field must be between 
+            0.8 and 1.0 (with 1.0 representing “black body” conditions). Default is .95.
+        min_stomatal_res: This field represents the resistance of the plants to moisture 
+            transport. It has units of s/m. Plants with low values of stomatal resistance 
+            will result in higher evapotranspiration rates than plants with high resistance. 
+            Values for this field must be in the range of 50.0 to 300.0. Default is 180.
+        soil_layer_name: This field is a unique reference name that the user assigns to the 
+            soil layer for a particular ecoroof. This name can then be referred to by 
+            other input data. Default is Green Roof Soil.
```

Comment:
I think we should not expose this property since it is not one that people can use from Honeybee.

In order to keep OpaqueConstructions as self-contained objects, we have users specify explicit EnergyMaterial layers for the construction and they are not allowed to pass the name/identifier of the layer like "Green Roof Soil". So I recommend removing this `soil_layer_name` as a set-able property from this class and we'll just always assume the EnergyPlus default value of "Green Roof Soil" when we serialize the material to and from IDF.

If you want to enable an EnergyPlus-type workflow for the soil layer in Honeybee, you can still have a read-only property on this class called `soil_layer`. But it should return an `EnergyMaterial` instance that has the same thickness, conductivity, etc. as the green roof material. You can generate it on the fly when the property is called such that, if the green roof material properties change, so do the soil layer properties. The identifier of this soil material should probably be derived from the `EnergyMaterialGreenRoof` identifier so that it doesn't conflict as easily with other possible green roof materials in the model.

This way, if people want to make a version of the construction that only has the soil without vegetation, they can easily do so.

Reply:
Sounds good! Thanks Chris! should be able to attend to this today/tomorrow

Reply:
OOOHHHHHH *facepalm*  #still_learning: I got this tho!